### PR TITLE
fix: relax dotenv dependency

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

* Relax dotenv dependency

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

## Further comments
n/a

### CODE REVIEW

File: autonomy/analyse/service.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # ------------------------------------------------------------------------------
 #
-#   Copyright 2023 Valory AG
+#   Copyright 2023-2024 Valory AG
```
Comment (0xArdi) [inline autonomy/analyse/service.py:4]: Why the change?

File: Pipfile [hunk 2]
```diff
@@ -41,7 +41,7 @@ hypothesis = "==6.21.6"
 protobuf = "<4.25.0,>=4.21.6"
 pytz = "==2022.2.1"
 py-ecc = "==6.0.0"
-python-dotenv = ">=0.14.5,<0.18.0"
+python-dotenv = ">=0.14.5,<0.22.0"
```
Comment (0xArdi) [inline Pipfile:44]: I think we will still need the open-aea release first. Otherwise, this wont take effect.

**Review by 0xArdi — COMMENTED** [review]
I think there are a bunch of accidental changes.

File: Pipfile (hunk 2 above)
Comment (dvilelaf) [reply Pipfile:44]: Yes, this is a WIP

File: autonomy/analyse/service.py (hunk 1 above)
Comment (dvilelaf) [reply autonomy/analyse/service.py:4]: This happened after running generators. Too many files changed.

**Review by angrybayblade — APPROVED** [review]
